File: nasa-apod-gallery/app/Models/ApodEntry.php
<?php
namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Carbon;

class ApodEntry extends Model
{
    use HasFactory;
    protected $table = 'apod_entries';
    protected $primaryKey = 'date';
    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [ /* Fields allowed for mass assignment */
        'date', 'title', 'explanation', 'url', 'hdurl',
        'media_type', 'copyright', 'service_version',
    ];

    // Accessor for is_image check
    public function getIsImageAttribute(): bool {
        return strtolower($this->media_type) === 'image';
    }
    // Accessor for year (optional)
    public function getYearAttribute(): ?int {
        try { return Carbon::parse($this->date)->year; } catch (\Exception $e) { return null; }
    }
    // Accessor for month (optional)
    public function getMonthAttribute(): ?int {
        try { return Carbon::parse($this->date)->month; } catch (\Exception $e) { return null; }
    }
    // Accessor for thumbnail URL (basic)
    public function getThumbnailUrlAttribute(): ?string {
        return $this->is_image ? $this->url : null;
    }
}

File: RestProject/RestProject/Models/ApodEntry.cs
using SQLite;
using System;

namespace RestProject.Models
{
    [Table("ApodEntries")]
    public class ApodEntry
    {
        [PrimaryKey]
        public string Date { get; set; }

        [Indexed]
        public string Title { get; set; }
        public string Explanation { get; set; }
        public string Url { get; set; }
        public string HdUrl { get; set; }
        public string MediaType { get; set; }
        public string Copyright { get; set; }
        public string ServiceVersion { get; set; }

        [Ignore]
        public int Year => !string.IsNullOrEmpty(Date) && DateTime.TryParse(Date, out var dt) ? dt.Year : 0;
        [Ignore]
        public int Month => !string.IsNullOrEmpty(Date) && DateTime.TryParse(Date, out var dt) ? dt.Month : 0;
        [Ignore]
        public bool IsImage => MediaType?.ToLowerInvariant() == "image";

        [Ignore]
        public string ThumbnailUrl => IsImage ? Url : string.Empty;
    }
}